packaging/macos/buildpackage.sh

```diff
@@ -0,0 +1,271 @@
+	exit 1
+fi
+
+command -v make >/dev/null 2>&1 || { echo >&2 "The OpenRA mod template requires make."; exit 1; }
+command -v python >/dev/null 2>&1 || { echo >&2 "The OpenRA mod template requires python."; exit 1; }
+
+require_variables() {
+	missing=""
+	for i in "$@"; do
+		eval check="\$$i"
+		[ -z "${check}" ] && missing="${missing}   ${i}\n"
+	done
+	if [ ! -z "${missing}" ]; then
+		echo "Required mod.config variables are missing:\n${missing}Repair your mod.config (or user.config) and try again."
+		exit 1
+	fi
+}
+
+if [ $# -eq "0" ]; then
+	echo "Usage: `basename $0` version [outputdir]"
+	exit 1
+fi
+
+PACKAGING_DIR=$(python -c "import os; print(os.path.dirname(os.path.realpath('$0')))")
+TEMPLATE_ROOT="${PACKAGING_DIR}/../../"
+ARTWORK_DIR="${PACKAGING_DIR}/../artwork/"
+
+# shellcheck source=mod.config
+. "${TEMPLATE_ROOT}/mod.config"
+
+if [ -f "${TEMPLATE_ROOT}/user.config" ]; then
+	# shellcheck source=user.config
+	. "${TEMPLATE_ROOT}/user.config"
+fi
+
+require_variables "MOD_ID" "ENGINE_DIRECTORY" "PACKAGING_DISPLAY_NAME" "PACKAGING_INSTALLER_NAME" \
+	"PACKAGING_OSX_LAUNCHER_TAG" "PACKAGING_OSX_LAUNCHER_SOURCE" "PACKAGING_OSX_LAUNCHER_TEMP_ARCHIVE_NAME" \
+	"PACKAGING_OSX_DMG_MOD_ICON_POSITION" "PACKAGING_OSX_DMG_APPLICATION_ICON_POSITION" "PACKAGING_OSX_DMG_HIDDEN_ICON_POSITION" \
+	"PACKAGING_FAQ_URL" "PACKAGING_OVERWRITE_MOD_VERSION"
+
+# Import code signing certificate
+if [ -n "${MACOS_DEVELOPER_CERTIFICATE_BASE64}" ] && [ -n "${MACOS_DEVELOPER_CERTIFICATE_PASSWORD}" ] && [ -n "${MACOS_DEVELOPER_IDENTITY}" ]; then
+	echo "Importing signing certificate"
+	echo "${MACOS_DEVELOPER_CERTIFICATE_BASE64}" | base64 --decode > build.p12
+	security create-keychain -p build build.keychain
+	security default-keychain -s build.keychain
+	security unlock-keychain -p build build.keychain
+	security import build.p12 -k build.keychain -P "${MACOS_DEVELOPER_CERTIFICATE_PASSWORD}" -T /usr/bin/codesign >/dev/null 2>&1
+	security set-key-partition-list -S apple-tool:,apple: -s -k build build.keychain >/dev/null 2>&1
+	rm -fr build.p12
+fi
+
+TAG="$1"
+if [ $# -eq "1" ]; then
+	OUTPUTDIR=$(python -c "import os; print(os.path.realpath('.'))")
+else
+	OUTPUTDIR=$(python -c "import os; print(os.path.realpath('$2'))")
+fi
+
+BUILTDIR="${PACKAGING_DIR}/build"
+PACKAGING_OSX_APP_NAME="OpenRA - ${PACKAGING_DISPLAY_NAME}.app"
+
+# Set the working dir to the location of this script
+cd "${PACKAGING_DIR}"
+
+modify_plist() {
+	sed "s|$1|$2|g" "$3" > "$3.tmp" && mv "$3.tmp" "$3"
+}
+
+echo "Building launcher"
+curl -s -L -o "${PACKAGING_OSX_LAUNCHER_TEMP_ARCHIVE_NAME}" -O "${PACKAGING_OSX_LAUNCHER_SOURCE}" || exit 3
+unzip -qq -d "${BUILTDIR}" "${PACKAGING_OSX_LAUNCHER_TEMP_ARCHIVE_NAME}"
+rm "${PACKAGING_OSX_LAUNCHER_TEMP_ARCHIVE_NAME}"
+
+modify_plist "{DEV_VERSION}" "${TAG}" "${BUILTDIR}/OpenRA.app/Contents/Info.plist"
+modify_plist "{FAQ_URL}" "${PACKAGING_FAQ_URL}" "${BUILTDIR}/OpenRA.app/Contents/Info.plist"
+
+pushd ${TEMPLATE_ROOT} > /dev/null
```

Comment:
We might want to wrap this into quotes, see https://github.com/OpenRA/ra2/pull/640/files#diff-a33916c985fb5ccaca70ed42f055e210R72.